site/gatsby-site/src/components/checklists/RiskSection.js

```diff
@@ -0,0 +1,447 @@
+    w-full px-2 
+    flex items-center
+
+    before:w-4 
+    before:pl-1 
+    before:bg-white 
+    before:text-lg 
+    ${props.generated ? 'before:text-gray-400' : 'before:text-red-700'}
+
+    ${props.className}
+  `,
+    }}
+  >
+    {props.children}
+  </summary>
+);
+
+const HeaderItemsGroup = (props) => (
+  <div
+    {...{
+      ...props,
+      className: `hidden md:flex px-2 gap-2 bg-white items-center ${props.className}`,
+    }}
+  >
+    {props.children}
+  </div>
+);
+
+const HeaderTextWithIcon = (props) => (
+  <div
+    {...{
+      ...props,
+      className: ` 
+    inline-flex flex gap-1 items-center
+    inline-block bg-${props.color || 'gray'}-200 px-3 rounded-lg 
+    text-${props.color || 'gray'}-800
+    ${props.className}
+  `,
+    }}
+  >
+    {props.children}
+  </div>
+);
+
+const PrecedentsQuery = (props) => (
+  <div {...{ ...props, className: `col-span-2 ${props.className}` }}>{props.children}</div>
+);
+
+const Precedents = (props) => (
+  <div {...{ ...props, className: `col-span-1 flex flex-col h-full ${props.className}` }}>
+    {props.children}
+  </div>
+);
+
+const RiskFields = (props) => (
+  <div {...{ ...props, className: `col-span-1 flex flex-col gap-2 ${props.className}` }}>
+    {props.children}
+  </div>
+);
+
+const PrecedentsList = (props) => (
+  <div
+    {...{
+      ...props,
+      className: `
+    flex flex-col gap-3 p-2  
+    h-full w-full max-w-full max-h-[30rem]
+    overflow-y-auto
+    bg-gray-100
+    border-1 border-gray-200 
+    rounded 
+    shadow-inner
+    ${props.className}
+  `,
+    }}
+  >
+    {props.children}
+  </div>
+);
+
+const updatePrecedents = async ({ risk, setPrecedents, allPrecedents }) => {
+  const updatedPrecedents = [];
+
+  for (const precedent of allPrecedents) {
+    for (const tag of precedent.tags) {
+      if ((risk.tags || []).includes(tag)) {
+        updatedPrecedents.push(precedent);
+        break;
+      }
+    }
+  }
+  setPrecedents(updatedPrecedents);
+};
+
+function ProgressCircle({ progress, className }) {
+  const r = 20;
+
+  const c = 2 * r * Math.PI;
+
+  const filledLength = c * progress;
+
+  const clearLength = c - filledLength;
+
+  return (
+    <div title={Math.round(progress * 100) + '%'} className={`${className} inline`}>
+      <svg width="20" height="20" viewBox="0 0 60 60">
+        <circle stroke="#d8dadc" strokeWidth="10" fill="transparent" r={r} cx="30" cy="30" />
+        <circle
+          stroke="#4084f8"
+          strokeWidth="10"
+          fill="transparent"
+          r={r}
+          cx="30"
+          cy="30"
+          strokeDasharray={`${filledLength} ${clearLength}`}
+          style={{ transform: 'rotate(-90deg)', transformOrigin: '50% 50%' }}
+        />
+      </svg>
+    </div>
+  );
+}
+
+const byNumPrecedents = (a, b) => a.precedents.length - b.precedents.length;
+
+const isNullish = (x) => [null, undefined, ''].includes(x);
+
+const byProperty = (p) => (a, b) => {
```

Comment:
I would highly appreciate a comment for this type of function, especially since it's used inline in multiple places and is really not trivial.

Reply:
Added!